Narrow spacing improvements

mike-spa opened the pull request:
Draft

RomanPudashkin commented on src/engraving/layout/layoutsystem.cpp:
```diff
@@ -1505,3 +1510,102 @@ void LayoutSystem::restoreTies(System* system)
     doLayoutTies(system, segList, stick, etick);
 }
+
+void LayoutSystem::manageNarrowSpacing(System* system, double& curSysWidth, double targetSysWidth)
```
This algorithm looks very suboptimal. We iterate over this system->measures() list many many times (including some implicit iterations, such as inside system->minSysTicks/maxSysTicks). Considering how slow the layout process is, we should at least do some performance tests before merging this

RomanPudashkin commented on src/engraving/layout/layoutsystem.cpp:
```diff
@@ -1505,3 +1510,102 @@ void LayoutSystem::restoreTies(System* system)
+    const Fraction minTicks = system->minSysTicks();
+    const Fraction maxTicks = system->maxSysTicks();
+    static constexpr double step = 0.2;
+    static constexpr double squeezeLimit = 0.3;
```
Please add a comment explaining why these constants have these values

mike-spa commented:
@RomanPudashkin yeah, I know it's extremely inefficient, but it's the only way to squeeze everything while still producing an acceptable engraving result. The good news is that `LayoutSystem::manageNarrowSpacing()` will _never_ be called by the default layout. The conditions for calling this function only happen if the user specifically requires to fit in the system more measures than can normally fit (again, also quite rare). In fact, I've tried the usual Beethoven 9 score and this function never gets called, so the final layout time is even slightly better than current master:

Master:
<img width="671" alt="master" src="https://user-images.githubusercontent.com/93707756/225945749-8de6b167-9ecf-471c-8ba2-126d6215d75c.png">

PR:
<img width="688" alt="thisPR" src="https://user-images.githubusercontent.com/93707756/225945778-a8431d3d-dd32-499d-a68f-acb0b6d98f66.png">

Besides that, I've removed the calls to minSysTicks and maxSysTicks cause I've realized that the values could be passed down by the caller. And I've added comments on the constants.

mike-spa commented:
Ok, I don't understand why the vtests seem to get stuck: I can run them locally on my machine (Windows) and they run fine. I'll try on my Linux machine and see if something changes.

mike-spa commented:
Update on vtests: they run perfectly fine both on my Windows machine and on my Linux machine. So I really don't know what to do with them here

igorkorsukov commented:
@mike-spa Please, disable (comment out) the drawdata steps so they don't block you (they are in alpha mode now), I'll see what's wrong later https://github.com/musescore/MuseScore/blob/master/.github/workflows/ci_vtests.yml#L153

mike-spa commented:
Ok, I've removed the problematic bit from the vtests and rebased it, let's hope it works 👍 (btw the current master crashes on startup)
